### ISSUE TITLE
Remove deprecated WINSOCK calls

### ISSUE THREAD

**Issue by Brett208** [OutpostUniverse/NetHelper#2]
VS2017's toolset is calling the functions inet_addr and inet_ntoa as deprecated. Should look at switching to different function calls to protect from future problems.

Warning	C4996	'inet_ntoa': Use inet_ntop() or InetNtop() instead or define _WINSOCK_DEPRECATED_NO_WARNINGS to disable deprecated API warnings	NetHelper	\nethelper\src\portforward.cpp	174	

Warning	C4996	'inet_addr': Use inet_pton() or InetPton() instead or define _WINSOCK_DEPRECATED_NO_WARNINGS to disable deprecated API warnings	NetHelper	\nethelper\src\portforward.cpp	258

**Comment by DanRStevens** [comment]
For reference:
- [inet_ntoa](https://linux.die.net/man/3/inet_ntoa)
- [inet_ntop](https://linux.die.net/man/3/inet_ntop)
- [inet_aton](https://linux.die.net/man/3/inet_aton) [not Posix, but common]
- [inet_pton](https://linux.die.net/man/3/inet_pton)

The newer method "p" methods extend address conversion to cover both IPv4 and IPv6. The `inet_ntop` converts from binary to string, while the reverse method `inet_pton` converts from string to binary.

This extension is part of Posix, and not simply a Microsoft extension. As such, it should be cross platform. The deprecation effectively extends to Linux as well.

----

I agree about upgrading these methods.

**Comment by Brett208** [comment]
For replacing inet_addr, how about:
``` C++
INT result = inet_pton(AF_INET, curAdapter->GatewayList.IpAddress.String, outGateway);
return result == 1;
```
I tested this and it seems to be returning reasonable results in a non error state.

---

For replacing inet_ntoa:
``` C++
char ipAddressStr[INET_ADDRSTRLEN];
strcpy_s(externalIp, sizeof(externalIp), 
    inet_ntop(AF_INET, &response.pnu.publicaddress.addr, &ipAddressStr[0], INET_ADDRSTRLEN));
```

I can't test this code right now because an error is encountered earlier in the code. I'm not on my home router so can test later.

-Brett

**Comment by DanRStevens** [comment]
For replacing `inet_addr` with `inet_pton`, your solution looks correct.

I would suggest removing the temporary value, by using:
```cpp
return inet_pton(AF_INET, curAdapter->GatewayList.IpAddress.String, outGateway) == 1;
```

----

I noticed `externalIp` (and `internalIp`) is statically sized to 46 bytes:
```cpp
char PortForwarder::internalIp[46] = {},
     PortForwarder::externalIp[46] = {};

```

We should probably update this to use the constant `INET6_ADDRSTRLEN`, as per the info in:
[Why is INET6_ADDRSTRLEN defined as 46 in C?](https://stackoverflow.com/questions/39443413/why-is-inet6-addrstrlen-defined-as-46-in-c)

And thank you for pointing out the IPv4 constant name. That made searching so much easier.

----

Comparing method signatures and documentation I linked earlier, I think we can remove the `strcpy_s` call, as this should be handled by the new method:
```cpp
char *inet_ntoa(struct in_addr in);
const char *inet_ntop(int af, const void *src, char *dst, socklen_t size);
```

Suggested code:
```cpp
if (!externalIp[0]) {
  inet_ntop(AF_INET, &response.pnu.publicaddress.addr, &externalIp[0], sizeof(externalIp));
}
```
